.github/workflows/ci.yml

```diff
@@ -5,20 +5,39 @@ on:
   push: { branches: [ master ] }
 
 jobs:
-  in-docker_test:
+  find_gradle_jobs:
     runs-on: ubuntu-18.04
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
     steps:
       - uses: actions/checkout@v2
-      - name: Build with Gradle
+      - id: set-matrix
+        run: echo "::set-output name=matrix::{\"gradle_args\":$(.github/workflows/scripts/find_gradle_jobs.sh)}"
```

Comment:
if you add the following task to Gradle:
```groovy
task taskMatrix {
    doLast {
        println(
            groovy.json.JsonOutput.toJson(subprojects.collect { it.tasks.getByName('check').path })
        )
    }
}
```

you can delegate the matrix creation to it:
```
echo "::set-output name=matrix::{\"gradle_args\":$(./gradlew --no-daemon -q taskMatrix)}"
```